src/colorizer/TimeControls.ts

```diff
@@ -8,23 +9,35 @@ export default class TimeControls {
   // TODO: Change to be React state?
   private timerId: number;
   private setFrameFn?: (frame: number) => void;
+  private playbackFps: number;
 
   private canvas: ColorizeCanvas;
   private isDisabled: boolean;
 
   private pauseCallbacks: (() => void)[];
 
-  constructor(canvas: ColorizeCanvas) {
+  constructor(canvas: ColorizeCanvas, playbackFps: number = DEFAULT_PLAYBACK_FPS) {
     this.canvas = canvas;
     this.timerId = DEFAULT_TIMER_ID;
     this.isDisabled = false;
     this.pauseCallbacks = [];
+    this.playbackFps = playbackFps;
   }
 
   public setFrameCallback(fn: (frame: number) => void): void {
     this.setFrameFn = fn;
   }
 
+  public setPlaybackFps(fps: number): void {
+    this.playbackFps = fps;
+    if (this.timerId !== DEFAULT_TIMER_ID) {
+      // restart playback with new fps
+      clearInterval(this.timerId);
+      this.pauseCallbacks.every((callback) => callback());
```

Comment:
`forEach` would make more semantic sense here than `every` and avoid a potential bug.

`every` is for checking that every element of an array satisfies some condition: it expects the function it receives to return a boolean, and returns the result of running that function on every element of the array and `&&`ing all the results. (It has a counterpart, `some`, which `||`s the results together. Other languages call these methods `all` and `any`, respectively.) But like the `&&` operator, `every` also *short-circuits*: once any predicate returns a falsy value (including `undefined`), it halts. So this method is likely to only run the first callback in `pauseCallbacks` and then stop, because it knows at that point that not every callback in the list returns `true`.

(Looks like this is a pattern established previously in this file? Sorry for not catching this before!)

Reply:
Thanks for the in-depth explanation! This is fixed :')